### PR TITLE
provide a human readable error message when download fails due to non purchase

### PR DESCRIPTION
**Pull request by pete-woods**

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -1142,7 +1142,12 @@ var download = func(name, downloadURL string, user *auth.UserState, s *Store, w
 		}
 		time.Sleep(time.Duration(n) * time.Millisecond)
 	}
-	if resp.StatusCode != 200 {
+	switch resp.StatusCode {
+	case http.StatusOK:
+		break
+	case http.StatusUnauthorized:
+		return fmt.Errorf("cannot download non-free snap without purchase")
```
Comment (mvo5) [inline store/store.go]: Do we need to consider i18n.G() here?

Comment (pete-woods) [inline store/store.go]: Well spotted

**Comment by mvo5** [comment]
:+1:

**Comment by mvo5** [comment]
Lets merge once tests are green.

**Pull request merged**